tests/cstor/volume/provision_negative_test.go

```diff
@@ -121,62 +133,84 @@ var _ = Describe("[Cstor Volume Provisioning Negative] Volume Provisioning", fun
+			}
+			By("Creating PVC, Desired Number of CVR Should Be Created")
+			ops.VerifyVolumeStatus(pvcObj,
+				cvrCount,
+				cvrPred,
+				cvPred,
+			)
 
 			// GetLatest PVC object
 			pvcObj, err = ops.PVCClient.
 				WithNamespace(nsObj.Name).
 				Get(pvcObj.Name, metav1.GetOptions{})
 			Expect(err).To(BeNil())
 		})
-	})
 
-	When("PVC Created During Error Injection", func() {
 		It("All Volume Replicas Should Not Be Healthy", func() {
-			By("Verify Volume Status after Injecting Errors", func() {
-				command := "zfs get guid | grep " + pvcObj.Spec.VolumeName + " | awk '{print $3}'"
-				_ = ops.ExecuteCMDEventually(
-					&poolPodList.Items[0],
-					"cstor-pool-mgmt", command, false)
-
-				// Eject the ZFSCreate error
-				injectError := debug.NewClient(hostIPPort)
-				err := injectError.PostInject(
-					debug.NewErrorInjection().
-						WithZFSCreateError(debug.Eject).
-						WithCVRUpdateError(debug.Inject))
-				Expect(err).To(BeNil())
-
-				//After ejecting the ZFS Create volume dataset should be created
-				zvolGUID := ops.ExecuteCMDEventually(
-					&poolPodList.Items[0],
-					"cstor-pool-mgmt", command, true)
-				Expect(zvolGUID).NotTo(BeEmpty())
-				// CVR Update will error due to error injection. So replica count
-				// should be n-1
-				ops.VerifyVolumeStatus(pvcObj, cstor.ReplicaCount-1)
-
-				//Eject CVRUpdate error then CVR should become Healthy
-				err = injectError.PostInject(
-					debug.NewErrorInjection().
-						WithCVRUpdateError(debug.Eject))
-				Expect(err).To(BeNil())
-
-				ops.VerifyVolumeStatus(pvcObj, cstor.ReplicaCount)
-			})
+			command := "zfs get guid | grep " + pvcObj.Spec.VolumeName + " | awk '{print $3}'"
+			_ = ops.ExecuteCMDEventually(
+				&poolPodList.Items[0],
+				"cstor-pool-mgmt", command, false)
+
+			// Eject the ZFSCreate error
+			injectError := debug.NewClient(hostIPPort)
+			err := injectError.PostInject(
+				debug.NewErrorInjection().
+					WithZFSCreateError(debug.Eject).
+					WithCVRUpdateError(debug.Inject))
+			Expect(err).To(BeNil())
+
+			//After ejecting the ZFS Create volume dataset should be created
+			zvolGUID := ops.ExecuteCMDEventually(
+				&poolPodList.Items[0],
+				"cstor-pool-mgmt", command, true)
+			Expect(zvolGUID).NotTo(BeEmpty())
+			// CVR Update will error due to error injection. So replica count
+			// should be n-1
+			By("Checking replica count after error injection")
+			cvrCount := cstor.ReplicaCount - 1
+			if cstor.ReplicaCount == 1 {
+				// Since replica count is 1 then we can check for 1 CVR only
+				// and that won't be in healthy state
+				cvrCount = 1
+			}
+			ops.VerifyVolumeStatus(pvcObj, cvrCount, cvrPred, cvPred)
+
+			//Eject CVRUpdate error then CVR should become Healthy
+			err = injectError.PostInject(
+				debug.NewErrorInjection().
+					WithCVRUpdateError(debug.Eject))
+			Expect(err).To(BeNil())
+
+			By("Checking replica count after error ejection")
+			ops.VerifyVolumeStatus(pvcObj, cstor.ReplicaCount, cvrPred, cvPred)
```

Comment:
When are we updating the predicate list? Now after ejecting the errors, we should update the predicate list.